src/pseudopeople/interface.py

```diff
@@ -170,18 +171,23 @@ def _reformat_dates_for_noising(data: pd.DataFrame, dataset: Dataset):
                 # re-parse the format string for each row
                 # https://github.com/pandas-dev/pandas/issues/44764
                 # Year is already guaranteed to be 4-digit: https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#timeseries-timestamp-limits
-                year_string = data[column].dt.year.astype(str)
-                month_string = _zfill_fast(data[column].dt.month.astype(str), 2)
-                day_string = _zfill_fast(data[column].dt.day.astype(str), 2)
+                is_na = data[column].isna()
+                data_column = data.loc[~is_na, column]
+                year_string = data_column.dt.year.astype(str)
+                month_string = _zfill_fast(data_column.dt.month.astype(str), 2)
+                day_string = _zfill_fast(data_column.dt.day.astype(str), 2)
                 if dataset.date_format == DATEFORMATS.YYYYMMDD:
-                    data[column] = year_string + month_string + day_string
+                    result = year_string + month_string + day_string
                 elif dataset.date_format == DATEFORMATS.MM_DD_YYYY:
-                    data[column] = month_string + "/" + day_string + "/" + year_string
+                    result = month_string + "/" + day_string + "/" + year_string
                 elif dataset.date_format == DATEFORMATS.MMDDYYYY:
-                    data[column] = month_string + day_string + year_string
+                    result = month_string + day_string + year_string
                 else:
                     raise ValueError(f"Invalid date format in {dataset.name}.")
 
+                data[column] = pd.Series(np.nan, dtype=str)
```

Comment:
Why do you need this line at all? Why can't you do:
```python
if dataset.date_format == DATEFORMATS.YYYYMMDD:
    data.loc[~is_na, column] = year_string + month_string + day_string
elif dataset.date_format == DATEFORMATS.MM_DD_YYYY:
        data.loc[~is_na, column] = month_string + "/" + day_string + "/" + year_string
...
```

Reply:
`.loc` does not change the dtype of the column if it can coerce whatever it gets to the dtype that the column already is. If I use your suggestion, the strings get parsed _back_ into dates when the column is currently a date (which is the case when there is no missingness in it).